ui/controlgroup.js

```diff
@@ -0,0 +1,175 @@
+		$.each( this.options.items, function( widget, selector ) {
+			that.element.children( selector ).map(function(){
+				return $( this )[ widget ]( "widget" )[ 0 ];
+			}).removeData( "ui-controlgroup-data" );
+		});
+		this._callChildMethod( "destroy" );
+		this.element.removeAttr( "role" );
+	},
+
+	_callChildMethod: function( method ) {
+		var that = this;
+		$.each( this.options.items, function( widget, selector ) {
+			var options = {};
+			if ( that[ "_" + widget + "_options" ] ) {
+				options = that[ "_" + widget + "_options" ]( "middle" );
+			}
+			if ( $.fn[ widget ] && selector ) {
+				that.element
+					.children( selector )[ widget ]( method ? method: options )
+					.each(function(){
+						if ( method !== "destroy" ) {
+							$( this )[ widget ]( "widget" ).data( "ui-controlgroup-data", {
+								"type": widget,
+								"element": $( this )
+							});
+						}
+					});
+			}
+		});
+	},
+
+	_button_options: function( position ) {
+		var cornerClasses = {
+			"middle": null,
+			"first": "ui-corner-" + ( ( this.options.direction === "vertical" )? "top" : "left" ),
+			"last": "ui-corner-" + ( ( this.options.direction === "vertical" )? "bottom" : "right" )
+		};
+
+		return {
+					classes: {
+						"ui-button": cornerClasses[ position ]
+					}
+				};
+	},
+
+	_checkboxradio_options: function( position ) {
+		var cornerClasses = {
+			"middle": null,
+			"first": "ui-corner-" + ( ( this.options.direction === "vertical" )? "top" : "left" ),
+			"last": "ui-corner-" + ( ( this.options.direction === "vertical" )? "bottom" : "right" )
+		};
+
+		return {
+					classes: {
+						"ui-checkboxradio-label": cornerClasses[ position ]
+					}
+				};
+	},
+
+	_selectmenu_options: function( position ) {
+		var classes = {
+			middle: {
+				"ui-selectmenu-button-open": null,
+				"ui-selectmenu-button-closed": null
+			},
+			first: {
+				"ui-selectmenu-button-open":
+					"ui-corner-" + ( ( this.options.direction === "vertical" )? "top": "tl" ),
+				"ui-selectmenu-button-closed":
+					"ui-corner-" + ( ( this.options.direction === "vertical" )? "top": "left" )
+			},
+			last: {
+				"ui-selectmenu-button-open":
+					( this.options.direction === "vertical" )? null: "ui-corner-tr",
+				"ui-selectmenu-button-closed":
+					"ui-corner-" + ( ( this.options.direction === "vertical" )? "bottom": "right" )
+			}
+
+		};
+		return { classes: classes[ position ] };
+	},
+
+	_setOption: function( key, value ) {
+		var original = this.options[ key ];
+
+		this._super( key, value );
+		if ( key === "direction" ) {
+			this.element.removeClass( "ui-controlgroup-" + original );
+		}
+		if ( key === "disabled" ) {
+			this._callChildMethod( value ? "disable" : "enable" );
+		} else {
+			this.refresh();
+		}
+
+	},
+
+	refresh: function() {
+		var children,
+			that = this;
+
+		this._addClass( "ui-controlgroup ui-controlgroup-" + this.options.direction );
+		this._callChildMethod();
+		children = this.element.children( ".ui-button" );
```

Comment:
Currently thats correct ( for whats being done here ) because we are finding all the visible buttons ( we want to ignore things like the actual input or selects ). However this will no longer be correct once we add textinput. Might need to put a little more thought into this. maybe just all visible children. I cant think of a case right off where that would not get the desired elements its just much slower.